lib/control-connection.js

```diff
@@ -282,6 +287,40 @@ ControlConnection.prototype.refreshHosts = function (newNodesUp, callback) {
         self.setPeersInfo(newNodesUp, err, result, next);
       });
     },
+    function resolveAndSetProtocolVersion(next) {
+      if (!self.initialized) {
+        // resolve protocol version from highest common version among hosts.
+        const highestCommon = self._highestCommonProtocolVersion(self.protocolVersion, self.hosts);
+        const reconnect = highestCommon !== self.protocolVersion;
+
+        // set protocol version on each host.
+        self.protocolVersion = highestCommon;
+        self.hosts.forEach(function (h) {
+          h.setProtocolVersion(self.protocolVersion);
+        });
+
+        // if protocol version changed, reconnect.
+        if (reconnect) {
+          self.log('info', 'Reconnecting since the protocol version changed to ' + highestCommon);
+          utils.series([
+            function closeConnection(cNext) {
+              self.host.removeFromPool(c, cNext);
```

Comment:
Thanks for the suggestion, will look into that.

Reply:
As far as I can tell, removing the listener for `socketClose` is not required, but i'll look more into that.  Instead of creating another function on `Host` i just inlined it into control connection (hope that's ok).